tests/infra/network.py

```diff
@@ -904,6 +904,36 @@ def wait_for_new_primary(self, old_primary, nodes=None, timeout_multiplier=2):
         flush_info(logs, None)
         raise error(f"A new primary was not elected after {timeout} seconds")
 
+    def wait_for_new_primary_in(
```

Comment:
Could we merge this with `wait_for_new_primary()` defined above?

Reply:
My intention is to replace all calls to wait_for_new_primary() with wait_for_new_primary_in() because it is more precise, in particular across reconfigurations, and then to drop the old call. I want to do that separately though.